### PR TITLE
Update Create LSP7 Guide

### PR DESCRIPTION
**Pull request by fhildeb**

### CODE REVIEW

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 1]
```diff
@@ -9,48 +9,65 @@ import TabItem from '@theme/TabItem';
...
 - [`LSP7Mintable`](../../contracts/contracts/LSP7DigitalAsset/presets/LSP7Mintable.md): allow creating new assets on the smart contract.
 - [`LSP7Burnable`](../../contracts/contracts/LSP7DigitalAsset/extensions/LSP7Burnable.md): allow tokens to be removed from the supply.
 
-You can modify the `mint()` function to adjust the amount of initially minted tokens and their receiver as described in the [LSP7 Mintable Documentation](../../contracts/contracts/LSP8IdentifiableDigitalAsset/presets/LSP8Mintable.md#mint).
+You can then import them within your Solidity contract file:
 
 ```solidity title="contracts/MyCustomToken.sol"
 // SPDX-License-Identifier: Apache-2.0
-pragma solidity ^0.8.4;
+pragma solidity ^0.8.9;
```
Comment (skimaharvey) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:43]: fine with me but any particular reason we bumped to `^0.8.9;`?

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 2]
```diff
@@ -62,217 +79,346 @@ contract CustomToken is LSP7Mintable, LSP7Burnable {
...
-### Create a deployment script
-
-Create a script to deploy the smart contract to the LUKSO Testnet network. You can either use a regular EOA (Externally Owned Account) or a Universal Profile.
+:::tip
 
-#### Deploy using a Universal Profile (Recommended)
+Add the `--verbose` and `--show-stack-traces` flags to get further debugging information and gas data.
 
-Deploy your contract using your Universal Profile. Make sure to have the [Universal Profile Browser Extension](/install-up-browser-extension) installed.
+This is especially important if a lot of functionality is inherited, as the bytecode might succeed the EVM limit of 24k bytes. You can find more information about optimization settings within the [Getting Started](./getting-started.md) section.
```
Comment (skimaharvey) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:113]: true

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 3]
```diff
@@ -62,217 +79,346 @@ contract CustomToken is LSP7Mintable, LSP7Burnable {
...
+  // Setup the provider
+  const provider = new ethers.JsonRpcProvider(
+    'https://rpc.testnet.lukso.gateway.fm',
+  );
+
+  // Setup the controller used to sign the deployment
+  const signer = new ethers.Wallet(process.env.PRIVATE_KEY as string, provider);
+  console.log(
+    'Deploying contracts with Universal Profile Controller: ',
+    signer.address,
```
Comment (skimaharvey) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:143]: we can do it this way but this we are in a hardhat environment 
may be we could use the ` const [deployer] = await ethers.getSigners()` as long as we have updated our `hardhat.config.ts` file

**Review by skimaharvey — COMMENTED** [review]
I think its a good first draft. 
May be we should benefit more from the the hardhat environment so that we dont need to import the rpc and deployer address in the script. 
Also lets see about the metadata part. you might be right that creating a separate section can be interesting

File: docs/learn/smart-contract-developers/create-lsp7-token.md (hunk 1 above)
Comment (fhildeb) [reply docs/learn/smart-contract-developers/create-lsp7-token.md:43]: Just to be in sync with the playground updates

File: docs/learn/smart-contract-developers/create-lsp7-token.md (hunk 3 above)
Comment (fhildeb) [reply docs/learn/smart-contract-developers/create-lsp7-token.md:143]: Thanks for this! Indeed we should use `ethers.getSigners()`- we already use it for EOA scripts, but just not for UPs... i simply copied it over from the `lukso-playground` - will change on both ends 👍🏻

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 4]
```diff
@@ -62,217 +79,200 @@ contract CustomToken is LSP7Mintable, LSP7Burnable {
...
     process.exit(1);
   });
 ```
 
-Deploy the contract:
+  </TabItem>
+
+</Tabs>
+
+If you have not yet setup the LUKSO networks and private keys in HardHat, please check out the previous [Getting Started](./getting-started.md) guide for smart contract developers. If you set up the Hardhat configuration, you can execute the deployment script using the following command:
```
Comment (Hugoo) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:248]: Hardhat

**Review by Hugoo — APPROVED** [review]
This LGTM but I didn't check the code. If you give me some time I can also check the code.

**Pull request merged**